ci: add bumpversion workflow

daniel-jones-dev opened the pull request:
Adds a manually-triggered workflow to bump the version, update the changelog, and open a PR for review.

## What it does

1. Runs `bump-my-version` to update version strings across source files (configured via `.bumpversion.toml`)
2. Updates `CHANGELOG.md` via `keep-a-changelog-action`
3. Opens a PR with the changes and extracted release notes for reviewer sign-off

Trigger via **Actions → Bump Version → Run workflow**, choosing `major`, `minor`, or `patch`.

## Security / reliability notes

- All third-party actions are pinned to full commit SHAs (supply-chain safety)
- Permissions are scoped to the job, not the workflow (`contents: write`, `pull-requests: write` only)
- `concurrency: group: bumpversion` prevents overlapping runs corrupting version state
- `peter-evans/create-pull-request` owns the commit — no manual `git commit` before the action runs
- `GITHUB_TOKEN` is used throughout; PRs created by this workflow will **not** trigger other `pull_request` workflows

## Prerequisite

Enable "Allow GitHub Actions to create and approve pull requests" in:
**Settings → Actions → General → Workflow permissions**

Commit:
ci: add bumpversion github workflow

ScottBrenner commented on .github/workflows/bumpversion.yml:
```diff
@@ -0,0 +1,152 @@
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
```
```suggestion
        uses: actions/checkout@v6
```

ScottBrenner commented on .github/workflows/run_ci.yml:
```diff
@@ -16,6 +16,19 @@ env:
+    name: Parse changelog to verify it is valid
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
```suggestion
      - uses: actions/checkout@v6
```

Commit:
ci: bumpversion action improvements

  1. Permissions — permissions: {} at workflow level, scoped contents: write + pull-requests: write to the job.
  2. Concurrency guard — group: bumpversion, cancel-in-progress: false so a second trigger waits rather than interrupting mid-bump.
  3. Pre-commit fix — removed the "Configure Git" and "Create version commit" steps; create-pull-request now owns the commit via commit-message.

Commit:
ci: bumpversion action - pin to specific hashes of 3rd party actions
